app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,610 @@
+                    defaultMessage: 'Loading Options...',
+                };
+                break;
+        }
+
+        return (
+            <View style={style.loadingContainer}>
+                <FormattedText
+                    {...text}
+                    style={style.loadingText}
+                />
+            </View>
+        );
+    }, [style, dataSource, loading, intl]);
+
+    const renderNoResults = useCallback((): JSX.Element | null => {
+        if (loading || page.current === INITIAL_PAGE) {
+            return null;
+        }
+
+        return (
+            <View style={style.noResultContainer}>
+                <FormattedText
+                    id='mobile.custom_list.no_results'
+                    defaultMessage='No Results'
+                    style={style.noResultText}
+                />
+            </View>
+        );
+    }, [loading, style]);
+
+    const renderChannelItem = (itemProps: any) => {
+        const itemSelected = Boolean(multiselectSelected[itemProps.item.id]);
+        return (
+            <ChannelListRow
+                key={itemProps.id}
+                {...itemProps}
+
+                theme={theme}
+                channel={itemProps.item as Channel}
+                selectable={isMultiselect || false}
+                selected={itemSelected}
+            />
+        );
+    };
+
+    const renderOptionItem = (itemProps: any) => {
+        const itemSelected = Boolean(multiselectSelected[itemProps.item.value]);
+        return (
+            <OptionListRow
+                key={itemProps.id}
+                {...itemProps}
+                theme={theme}
+                selectable={isMultiselect || false}
+                selected={itemSelected}
+            />
+        );
+    };
+
+    const renderUserItem = (itemProps: any): JSX.Element => {
+        const itemSelected = Boolean(multiselectSelected[itemProps.item.id]);
+
+        return (
+            <UserListRow
+                key={itemProps.id}
+                {...itemProps}
+                theme={theme}
+                selectable={isMultiselect || false}
+                user={itemProps.item}
+                teammateNameDisplay={teammateNameDisplay}
+                selected={itemSelected}
+            />
+        );
+    };
+
+    const getRenderItem = useCallback((): (itemProps: any) => JSX.Element => {
+        switch (dataSource) {
+            case ViewConstants.DATA_SOURCE_USERS:
+                return renderUserItem;
+            case ViewConstants.DATA_SOURCE_CHANNELS:
+                return renderChannelItem;
+            default:
+                return renderOptionItem;
+        }
+    }, [dataSource, renderUserItem, renderChannelItem, renderOptionItem]);
+
+    const renderSelectedOptions = useCallback((): React.ReactElement<any, string> | null => {
+        const selectedItems: any = Object.values(multiselectSelected);
+
+        let optionComponents: React.ReactElement<any, string> | null = null;
+
```

Comment:
Nit: this is just about taste, so feel free to leave it as it is if you like it better, but I find this would be simpler / prettier if you remove the variable, and just do 
```
if (!data.length) {return null;}
return (<>
    ...
</>);
```